dist/src/main/java/io/camunda/application/commons/job/JobHandlerConfiguration.java

```diff
@@ -13,23 +13,20 @@
 import io.camunda.zeebe.gateway.impl.job.LongPollingActivateJobsHandler;
 import io.camunda.zeebe.gateway.impl.job.RoundRobinActivateJobsHandler;
 import io.camunda.zeebe.gateway.protocol.rest.JobActivationResponse;
+import io.camunda.zeebe.gateway.rest.ConditionalOnRestGatewayEnabled;
 import io.camunda.zeebe.gateway.rest.ResponseMapper;
 import io.camunda.zeebe.gateway.rest.controller.JobActivationRequestResponseObserver;
 import io.camunda.zeebe.gateway.rest.controller.ResponseObserverProvider;
 import io.camunda.zeebe.scheduler.Actor;
 import io.camunda.zeebe.scheduler.ActorScheduler;
 import java.util.concurrent.CompletableFuture;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.util.unit.DataSize;
 
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnProperty(
-    name = {"zeebe.broker.gateway.enable", "camunda.rest.enabled"},
-    havingValue = "true",
-    matchIfMissing = true)
+@ConditionalOnRestGatewayEnabled
```

Comment:
This does not ensure that the beans/configurations are resolved in the expected order in Spring. This condition may be evaluated as `true` because the bean `RestGatewayDisabled` is missing at that time in moment when registering the configuration/bean, meaning Spring will initiate that configuration `JobHandlerConfiguration` eventually. But the bean `RestGatewayDisabled` might created as a next step. In that case, the bean `JobHandlerConfiguration` will be initiated even when the bean `RestGatewayDisabled` was initiated too. (In same cases, Spring might do the scanning in a different order, which will result in the expected result.)

That is why `@ConditionalOnProperty` has been used so far at these places. An alternative would be that `ConditionalOnRestGatewayEnabled` checks the configuration properties as well to have a deterministic behavior.

Reply:
Ohhhhh, OK, I see. Yeah that makes, now that you mentioned it, I remember this general issue with Spring.

The only problem I see is that the usage of the properties is not entirely correct. The `zeebe.broker.gateway.enable` property is irrelevant for the gateway scenario. Even if I set this to `false`, the REST controllers should work in this use case but they currently don't. I guess that is what you mentioned as an edge case in the last PR you created touching this code. Now I really understand what you meant, I guess.

The inherent problem is that the `ConditionalOnRestGatewayEnabled` doesn't know which use case it is used in, gateway or broker. Thus it cannot decide which properties to check.

I guess we can live with this edge case but I find that a bit irritating since it's spread all over the code.

Reply:
I will move the propery checks into the `ConditionalOnRestGatewayEnabled` and `ConditionalOnRestQueryEnabled` annotations then to at least be consistent. There have been inconsistencies already. We have a central place to determine if the REST API is enabled then.

Reply:
I provided a simplified way of evaluating the REST API properties with https://github.com/camunda/camunda/pull/20584/commits/02e8c278b2672d1a6d905be7177502c7f5fe8b24. It includes disabling the REST API if `zeebe.broker.gateway.enable` is set to false in all scenarios, which is not entirely correct but also reflects the existing behavior. In my regard, it makes our property handling more consistent and centralized, thus is should be less error-prone.